src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -0,0 +1,55 @@
+package com._up.megastore.services.implementations;
+
+import com._up.megastore.controllers.requests.CreateUserRequest;
+import com._up.megastore.controllers.responses.UserResponse;
+import com._up.megastore.data.model.User;
+import com._up.megastore.data.repositories.IUserRepository;
+import com._up.megastore.services.interfaces.IMailSenderService;
+import com._up.megastore.services.interfaces.IUserService;
+import com._up.megastore.services.mappers.UserMapper;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService implements IUserService {
+
+  private final IMailSenderService mailSenderService;
+  private final IUserRepository userRepository;
+  private final PasswordEncoder passwordEncoder;
+
+  public UserService(IUserRepository userRepository, PasswordEncoder passwordEncoder,
+      IMailSenderService mailSenderService) {
+    this.userRepository = userRepository;
+    this.passwordEncoder = passwordEncoder;
+    this.mailSenderService = mailSenderService;
+  }
+
+  @Override
+  public UserResponse saveUser(CreateUserRequest createUserRequest) {
+    validateRequest(createUserRequest);
+
+    User user = UserMapper.toUser(createUserRequest);
+    String passwordEncoded = passwordEncoder.encode(createUserRequest.password());
+    user.setPassword(passwordEncoded);
+    mailSenderService.sendWelcomeEmail(user);
```

Comment:
Siguiendo el comentario anterior, acá no estamos catcheando una posible excepcion que lance el service, lo que podria dar un error y hacer que el usuario no registre realmente solo porque no se mandó el mail.

Queremos evitar que el usuario se registre si nosotros tenemos un problema a la hora de enviar el mail? A mi en lo personal registraría de todas formas y si no se envió el mail bueno, cositas que pasan.

Eso lo podemos hacer haciendo esto

```suggestion
    String passwordEncoded = passwordEncoder.encode(createUserRequest.password());
    user.setPassword(passwordEncoded);
    try {
      mailSenderService.sendWelcomeEmail(user);
    } catch (MessagingException exception) {
      // Se loggea que no se pudo enviar el mail y sigue la ejecución del método
    }
```